sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -0,0 +1,248 @@
+  private final Matcher serviceTypeMatcher;
+  private final Matcher resourceArnMatcher;
+
+  SamplingRuleApplier(GetSamplingRulesResponse.SamplingRule rule) {
+    fixedRateSampler = Sampler.parentBased(Sampler.traceIdRatioBased(rule.getFixedRate()));
+
+    if (rule.getAttributes().isEmpty()) {
+      attributeMatchers = Collections.emptyMap();
+    } else {
+      attributeMatchers =
+          rule.getAttributes().entrySet().stream()
+              .collect(Collectors.toMap(Map.Entry::getKey, e -> toMatcher(e.getValue())));
+    }
+
+    urlPathMatcher = toMatcher(rule.getUrlPath());
+    serviceNameMatcher = toMatcher(rule.getServiceName());
+    httpMethodMatcher = toMatcher(rule.getHttpMethod());
+    hostMatcher = toMatcher(rule.getHost());
+    serviceTypeMatcher = toMatcher(rule.getServiceType());
+    resourceArnMatcher = toMatcher(rule.getResourceArn());
+  }
+
+  boolean matches(String name, Attributes attributes, Resource resource) {
+    int matchedAttributes = 0;
+    String httpTarget = null;
+    String httpMethod = null;
+    String host = null;
+
+    for (Map.Entry<AttributeKey<?>, Object> entry : attributes.asMap().entrySet()) {
+      if (entry.getKey().equals(SemanticAttributes.HTTP_TARGET)) {
+        httpTarget = (String) entry.getValue();
+      } else if (entry.getKey().equals(SemanticAttributes.HTTP_METHOD)) {
+        httpMethod = (String) entry.getValue();
+      } else if (entry.getKey().equals(SemanticAttributes.HTTP_HOST)) {
+        host = (String) entry.getValue();
+      }
+
+      Matcher matcher = attributeMatchers.get(entry.getKey().getKey());
+      if (matcher == null) {
+        continue;
+      }
+      if (matcher.matches(entry.getValue().toString())) {
+        matchedAttributes++;
+      } else {
+        return false;
+      }
+    }
+    // All attributes in the matched attributes must have been present in the span to be a match.
+    if (matchedAttributes != attributeMatchers.size()) {
+      return false;
+    }
+
+    return urlPathMatcher.matches(httpTarget)
+        && serviceNameMatcher.matches(name)
+        && httpMethodMatcher.matches(httpMethod)
+        && hostMatcher.matches(host)
+        && serviceTypeMatcher.matches(getServiceType(resource))
+        && resourceArnMatcher.matches(getArn(attributes, resource));
+  }
+
+  SamplingResult shouldSample(
+      Context parentContext,
+      String traceId,
+      String name,
+      SpanKind spanKind,
+      Attributes attributes,
+      List<LinkData> parentLinks) {
+    return fixedRateSampler.shouldSample(
+        parentContext, traceId, name, spanKind, attributes, parentLinks);
+  }
+
+  @Nullable
+  private static String getArn(Attributes attributes, Resource resource) {
+    String arn = resource.getAttributes().get(ResourceAttributes.AWS_ECS_CONTAINER_ARN);
+    if (arn != null) {
+      return arn;
+    }
+    String cloudPlatform = resource.getAttributes().get(ResourceAttributes.CLOUD_PLATFORM);
```

Comment:
NIT: rest of the method could be factored out into a helper "getLambdaArn" call for improved readability.